src/theory/arith/nl/nl_lemma_utils.cpp

```diff
@@ -21,6 +21,12 @@ namespace theory {
 namespace arith {
 namespace nl {
 
+std::ostream& operator<<(std::ostream& out, NlLemma& n)
+{
+  out << n.d_lemma;
+  return out;
```

Comment:
It is this both returned and moved to the parameter?

Reply:
This is the common way of writing to streams, for another example, see: https://github.com/CVC4/CVC4/blob/master/src/expr/proof_rule.cpp#L96.